src/Illuminate/Support/Collection.php

```diff
@@ -870,6 +870,19 @@ protected function valueRetriever($value)
     }
 
     /**
+     * Walk over each of the items.
+     *
+     * @param  callable  $callback
+     * @return static
+     */
+    public function walk(callable $callback)
+    {
+        array_walk($this->items, $callback);
```

Comment:
this looks wrong - you're modifying the original collection as well as the new one